lib/plugins/aws/lib/updateStack.js

```diff
@@ -45,7 +46,17 @@ module.exports = {
     return this.provider.request('CloudFormation',
       'createStack',
       params)
-      .then((cfData) => this.monitorStack('create', cfData));
+      .then((cfData) => this.monitorStack('create', cfData)).catch((e) => {
+        if (this.serverless.service.deployment && this.serverless.service.deployment.deploymentId) {
+          const deploymentData = this.serverless.service.deployment;
+          deploymentData.status = 'Failed';
+          return platform.createDeployment(deploymentData)
```

Comment:
Same here. Shouldn't we create a new deployment beforehand with the status `in-progress` (this is the default one if no status is provided) and then update it here with the status of `failed`?